alacritty_terminal/src/term/render.rs

```diff
@@ -223,6 +230,10 @@ impl RenderableCell {
             bg_rgb = config_bg.color(fg_rgb, bg_rgb);
             fg_rgb = selected_fg;
 
+            if cell.ul().is_none() {
+                ul_rgb = fg_rgb;
+            }
```

Comment:
Why not just put the entire underline logic below these conditionals? That way it shouldn't be necessary to mutate `ul_rgb` after the fact.